src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -56,6 +73,8 @@ public ODataSerializerContext(ResourceContext resource, SelectExpandClause selec
 
             ExpandedResource = resource; // parent resource
             SelectExpandClause = selectExpandClause;
+            QueryOptions = queryOptions;
+            ExpandedSelectItem = expandedItem;
```

Comment:
setting queryOptions should set expandItem in HttpRequestMessageProperties, so we should only have to set this if queryOptions is not specified.